src/common/utils/getHydratedPieces.js

```diff
@@ -1,27 +1,22 @@
-import {
-  batchFetch,
-  ITEM_STATUS,
-} from '@folio/stripes-acq-components';
+import { ITEM_STATUS } from '@folio/stripes-acq-components';
 
 export const getPieceStatusFromItem = (item) => {
   return item?.status?.name || ITEM_STATUS.undefined;
 };
 
-export function getHydratedPieces(pieces, mutatorRequests, mutatorItems) {
-  // TODO: fetch consortium context's data
-  const itemsIds = pieces.filter(({ itemId }) => itemId).map(({ itemId }) => itemId);
-  const requestsPromise = batchFetch(mutatorRequests, pieces, (piecesChunk) => {
-    const itemIdsQuery = piecesChunk
-      .filter(piece => piece.itemId)
-      .map(piece => `itemId==${piece.itemId}`)
-      .join(' or ');
-
-    return itemIdsQuery ? `(${itemIdsQuery}) and status="Open*"` : '';
-  });
+export function getHydratedPieces({
+  crossTenant,
+  fetchPieceItems,
+  fetchPieceRequests,
+  isConsortium,
+  pieces,
+}) {
+  const mutatorItems = fetchPieceItems({ pieces, crossTenant, isConsortium });
+  const mutatorRequests = fetchPieceRequests({ pieces });
 
   return Promise.all([
-    batchFetch(mutatorItems, itemsIds),
-    requestsPromise,
+    mutatorItems,
+    mutatorRequests,
```

Comment:
Could you please put requests directly here or not use `mutator` in the consts name? Because `mutator` is an old thing that referenced to `stripes-connect` mutators